src/main/webapp/js/instructorCourseEdit.js

```diff
@@ -197,22 +210,27 @@ function checkPrivilegesOfTutorForInstructor(instrNum) {
 
 function checkPrivilegesOfCustomForInstructor(instrNum) {
 	var numOfInstr = $("form[id^='formEditInstructor']").length;
-	if (instrNum <= numOfInstr && INSTRUCTOR_COURSE_EDIT_INSTRUCTOR_ACCESS_LEVEL_WHEN_LOADING_PAGE.length >= instrNum 
-			&& INSTRUCTOR_COURSE_EDIT_INSTRUCTOR_ACCESS_LEVEL_WHEN_LOADING_PAGE[instrNum-1] === "Custom") {
-		$("#tunePermissionsDivForInstructor" + 1 + " input[checked='checked']").prop("checked", true);
-		$("#tunePermissionsDivForInstructor" + 1 + " input[checked!='checked']").prop("checked", false);
+	var originalRole = instructorCourseEditInstructorAccessLevelWhenLoadingPage[instrNum-1];
+	
+	if ((instrNum <= numOfInstr) && (instrNum <= instructorCourseEditInstructorAccessLevelWhenLoadingPage.length)) {
+		if (originalRole === 'Custom') {
+			// 'Custom' => 'Not Custom' => 'Custom'
+			// restore old values
+			for (var i = 0; i < instructorPrivilegeValues.length; i++) {
+				var privilege = instructorPrivilegeValues[i];
+				var valueToSet = instructorCourseEditDefaultPrivilegeValuesForCustomRole[instrNum][privilege];
+				$("#tunePermissionsDivForInstructor" + instrNum + " input[name='" + privilege + "']").prop("checked", valueToSet);
+			}
+		} else {
+			// 'Not Custom' => 'Custom'
+			checkPrivilegesOfRoleForInstructor(instrNum, originalRole);
+		}
 	} else {
-		$("#tunePermissionsDivForInstructor" + instrNum + " input[name='canmodifycourse']").prop("checked", false);
-		$("#tunePermissionsDivForInstructor" + instrNum + " input[name='canmodifyinstructor']").prop("checked", false);
-		$("#tunePermissionsDivForInstructor" + instrNum + " input[name='canmodifysession']").prop("checked", false);
-		$("#tunePermissionsDivForInstructor" + instrNum + " input[name='canmodifystudent']").prop("checked", false);
-		$("#tunePermissionsDivForInstructor" + instrNum + " input[name='canviewstudentinsection']").prop("checked", false);
-		$("#tunePermissionsDivForInstructor" + instrNum + " input[name='cangivecommentinsection']").prop("checked", false);
-		$("#tunePermissionsDivForInstructor" + instrNum + " input[name='canviewcommentinsection']").prop("checked", false);
-		$("#tunePermissionsDivForInstructor" + instrNum + " input[name='canmodifycommentinsection']").prop("checked", false);
-		$("#tunePermissionsDivForInstructor" + instrNum + " input[name='cansubmitsessioninsection']").prop("checked", false);
-		$("#tunePermissionsDivForInstructor" + instrNum + " input[name='canviewsessioninsection']").prop("checked", false);
-		$("#tunePermissionsDivForInstructor" + instrNum + " input[name='canmodifysessioncommentinsection']").prop("checked", false);
+		// New Instructor?
+		// Custom role's privilege will be empty
+		for (var i = 0; i < instructorPrivilegeValues.length; i++) {
+			$("#tunePermissionsDivForInstructor" + instrNum + " input[name='" + instructorPrivilegeValues[i] + "']").prop("checked", false);
```

Comment:
You can refer to #3946. It's supposed to have no privilege when we set Custom for new instructors.